### PR TITLE
[1.4.x] Improve support for Python 3.12

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

1.4.x backport of #164 plus other fixes required for Python 3.12

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
@xhochy, we can try this backport directly on the feedstock as a [patch](https://patch-diff.githubusercontent.com/raw/conda/menuinst/pull/165.patch). If that works, we can cut a patch release for the 1.4.x branch.

File: menuinst/winshortcut.cpp [hunk 1]
```diff
@@ -30,18 +32,24 @@
 
 static PyObject *CreateShortcut(PyObject *self, PyObject *args)
 {
-    Py_UNICODE *path; /* path and filename */
-    Py_UNICODE *description;
-    Py_UNICODE *filename;
+    PyObject *py_path; /* path and filename */
```
Comment (xhochy) [inline menuinst/winshortcut.cpp:35]: Don't we need to free (as in `Py_DECREF `) these objects at the end?

Comment (jaimergp) [reply menuinst/winshortcut.cpp:35]: I tried in #164 but it crashes. According to @ngoldbaum, these are borrowed references so they don't need decref'ing.

**Review by xhochy — APPROVED** [review]
Looking good!

File: menuinst/winshortcut.cpp (hunk 1 above)
Comment (ngoldbaum) [reply menuinst/winshortcut.cpp:35]: in the docs for PyArg_ParseTuple, under the `O` code:

> The C program thus receives the actual object that was passed. A new strong reference to the object is not created (i.e. its reference count is not increased).

`U` is the same as `O` except is does a PyUnicode_Check as well.

**Review by jezdez — APPROVED** [review]

**Pull request merged**